test/ui-test/src/screens/SettingsScreen.py

```diff
@@ -279,32 +283,83 @@ def set_bio(self, bio: str):
+        github = ""
+        youtube = ""
+        discord = ""
+        telegram = ""
+        custom_link_text = ""
+        custom_link = ""
+        
+        if table is not None:
+            verify_equals(8, len(table)) # Expecting 8 as social media link fields to verify
+            twitter = table[0][0]
+            personal_site = table[1][0]
+            github = table[2][0]
+            youtube = table[3][0]
+            discord = table[4][0]
+            telegram = table[5][0]
+            custom_link_text = table[6][0]
+            custom_link = table[7][0]       
+        
         click_obj_by_name(ProfileSettingsScreen.OPEN_SOCIAL_LINKS_DIALOG.value)
 
         click_obj_by_name(ProfileSettingsScreen.TWITTER_SOCIAL_LINK_IN_DIALOG.value)
-        verify(type(ProfileSettingsScreen.TWITTER_SOCIAL_LINK_IN_DIALOG.value, twitter), "set twitter")
-
+        verify(setText(ProfileSettingsScreen.TWITTER_SOCIAL_LINK_IN_DIALOG.value, twitter), "set twitter")
         click_obj_by_name(ProfileSettingsScreen.PERSONAL_SITE_LINK_IN_DIALOG.value)
-        verify(type(ProfileSettingsScreen.PERSONAL_SITE_LINK_IN_DIALOG.value, personal_site), "set personal site")
-
+        verify(setText(ProfileSettingsScreen.PERSONAL_SITE_LINK_IN_DIALOG.value, personal_site), "set personal site")
+        click_obj_by_name(ProfileSettingsScreen.GITHUB_SOCIAL_LINK_IN_DIALOG.value)
+        verify(setText(ProfileSettingsScreen.GITHUB_SOCIAL_LINK_IN_DIALOG.value, github), "set github")
+        click_obj_by_name(ProfileSettingsScreen.YOUTUBE_SOCIAL_LINK_IN_DIALOG.value)
+        verify(setText(ProfileSettingsScreen.YOUTUBE_SOCIAL_LINK_IN_DIALOG.value, youtube), "set youtube")
+        click_obj_by_name(ProfileSettingsScreen.DISCORD_SOCIAL_LINK_IN_DIALOG.value)
+        verify(setText(ProfileSettingsScreen.DISCORD_SOCIAL_LINK_IN_DIALOG.value, discord), "set discord")
+        click_obj_by_name(ProfileSettingsScreen.TELEGRAM_SOCIAL_LINK_IN_DIALOG.value)
+        verify(setText(ProfileSettingsScreen.TELEGRAM_SOCIAL_LINK_IN_DIALOG.value, telegram), "set telegram")
         click_obj_by_name(ProfileSettingsScreen.CUSTOM_LINK_IN_DIALOG.value)
-        verify(type(ProfileSettingsScreen.CUSTOM_LINK_IN_DIALOG.value, custom_link_name), "set custom link name")
-
+        verify(setText(ProfileSettingsScreen.CUSTOM_LINK_IN_DIALOG.value, custom_link_text), "set custom link name")
         click_obj_by_name(ProfileSettingsScreen.CUSTOM_URL_IN_DIALOG.value)
-        verify(type(ProfileSettingsScreen.CUSTOM_URL_IN_DIALOG.value, custom_link), "set custom link url")
+        verify(setText(ProfileSettingsScreen.CUSTOM_URL_IN_DIALOG.value, custom_link), "set custom link url")
 
         click_obj_by_name(ProfileSettingsScreen.CLOSE_SOCIAL_LINKS_DIALOG.value)
         click_obj_by_name(SettingsScreenComponents.SAVE_BUTTON.value)
 
-    def verify_social_links(self, twitter, personal_site, custom_link_name, custom_link: str):
+    def verify_social_links(self, table):
```

Comment:
This function does something very similar to `set_social_links` at the top. Maybe you could call `set_social_links` for the first operations?

Reply:
Oh right, it's Python, in my head those were some globals or something, but now, there is no keyword to set a local variable lol.
Ignore my comment.